crates/trie/trie/src/state.rs

```diff
@@ -25,11 +25,11 @@ impl HashedPostState {
     /// Initialize [`HashedPostState`] from bundle state.
     /// Hashes all changed accounts and storage entries that are currently stored in the bundle
     /// state.
-    pub fn from_bundle_state<'a>(
-        state: impl IntoParallelIterator<Item = (&'a Address, &'a BundleAccount)>,
-    ) -> Self {
+    pub fn from_bundle_state(state: &BundleState) -> Self {
         let hashed = state
-            .into_par_iter()
+            .state
+            .iter()
+            .par_bridge()
```

Comment:
Is there a specific reason you have changed this implementation? This has implications for the state root work I'm working on. I think we should avoid changes to core data structures unless they are necessary.